File: CSharp/StringCalculator.Logic.Tests/StringCalculator2Test.cs
using System;
using NUnit.Framework;

namespace StringCalculator.Logic.Tests
{
    public class StringCalculator2Test
    {
        [Test]
        public void whenMoreThan2NumbersAreUsedThenExceptionIsThrown()
        {
            Assert.Throws(typeof(ArgumentException), () => StringCalculator2.Add("1,2,3"));
        }

        [Test]
        public void when2NumbersAreUsedThenNoExceptionIsThrown()
        {
            Assert.DoesNotThrow(() => StringCalculator2.Add("1,2"));
        }

        [Test]
        public void whenNonNumberIsUsedThenExceptionIsThrown()
        {
            Assert.Throws(typeof(FormatException), () => StringCalculator2.Add("1,x"));
        }

        [Test]
        public void whenEmptyStringIsUsedThenReturnValueIs0()
        {
            Assert.AreEqual(0, StringCalculator2.Add(""));
        }
    }
}

File: Java/StringCalculator/src/test/java/StringCalculator2Test.java
import org.junit.Assert;
import org.junit.Test;

public class StringCalculator2Test {

	@Test(expected = RuntimeException.class)
	public void whenMoreThan2NumbersAreUsedThenExceptionIsThrown() {
		StringCalculator2.add("1,2,3");
	}
	
	@Test
	public void when2NumbersAreUsedThenNoExceptionIsThrown() {
		StringCalculator2.add("1,2");
		Assert.assertTrue(true);
	}

	@Test(expected = RuntimeException.class)
	public void whenNonNumberIsUsedThenExceptionIsThrown() {
		StringCalculator2.add("1,X");
	}
	
	@Test
	public void whenEmptyStringIsUsedThenReturnValueIs0() {
		Assert.assertEquals(0, StringCalculator2.add(""));
	}


}
